controller/controller_server/utils.py

```diff
@@ -192,15 +192,26 @@ def generate_csi_publish_volume_response(lun, connectivity_type, config, array_i
 
     lun_param = config["controller"]["publish_context_lun_parameter"]
     connectivity_param = config["controller"]["publish_context_connectivity_parameter"]
-    hash_by_connectivity = {
-        'iscsi': config["controller"]["publish_context_array_iqn"],
-        'fc': config["controller"]["publish_context_fc_initiators"]}
-
-    array_initiators = ",".join(array_initiators)
-    res = csi_pb2.ControllerPublishVolumeResponse(
-        publish_context={lun_param: str(lun),
-                         connectivity_param: connectivity_type,
-                         hash_by_connectivity[connectivity_type]: array_initiators})
+    separator = config["controller"]["publish_context_separator"]
+
+    publish_context = {
+        lun_param: str(lun),
+        connectivity_param: connectivity_type
+    }
+
+    if connectivity_type == ISCSI_CONNECTIVITY_TYPE:
+        for iqn, ips in array_initiators.items():
+            publish_context[iqn] = separator.join(ips)
+
+        array_initiators_param = config["controller"]["publish_context_array_iqn"]
+        publish_context[array_initiators_param] = separator.join(array_initiators.keys())
+    elif connectivity_type == FC_CONNECTIVITY_TYPE:
+        array_initiators_param = config["controller"]["publish_context_fc_initiators"]
+        publish_context[array_initiators_param] = separator.join(array_initiators)
+    else:
+        raise UnsupportedConnectivityTypeError(connectivity_type)
```

Comment:
Again,  this error should be raise before calling generate_xxx, no errors can be raise in this method.